code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -51,91 +60,162 @@
 		return
-	active = FALSE
-	icon_state = "reactiveoff"
-	inhand_icon_state = "reactiveoff"
-	reactivearmor_cooldown = world.time + 200
+	visible_message(emp_message)
+	bad_effect = TRUE
+	addtimer(VARSET_CALLBACK(src, bad_effect, FALSE), 30 SECONDS)
 
 //When the wearer gets hit, this armor will teleport the user a short distance away (to safety or to more danger, no one knows. That's the fun of it!)
 /obj/item/clothing/suit/armor/reactive/teleport
 	name = "reactive teleport armor"
 	desc = "Someone separated our Research Director from his own head!"
+	emp_message = "<span class='warning'>The reactive armor's teleportation calculations begin spewing errors!</span>"
+	cooldown_message = "<span class='danger'>The reactive teleport system is still recharging! It fails to activate!</span>"
+	reactivearmor_cooldown_duration = 10 SECONDS
 	var/tele_range = 6
 	var/rad_amount= 15
-	reactivearmor_cooldown_duration = 100
 
-/obj/item/clothing/suit/armor/reactive/teleport/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
-	if(!active)
-		return FALSE
-	if(prob(hit_reaction_chance))
-		if(world.time < reactivearmor_cooldown)
-			owner.visible_message("<span class='danger'>The reactive teleport system is still recharging! It fails to teleport [owner]!</span>")
-			return FALSE
-		owner.visible_message("<span class='danger'>The reactive teleport system flings [owner] clear of [attack_text], shutting itself off in the process!</span>")
-		playsound(get_turf(owner),'sound/magic/blink.ogg', 100, TRUE)
-		do_teleport(owner, get_turf(owner), tele_range, no_effects = TRUE, channel = TELEPORT_CHANNEL_BLUESPACE)
-		owner.rad_act(rad_amount)
-		reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
-		return TRUE
-	return FALSE
+/obj/item/clothing/suit/armor/reactive/teleport/reactive_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>The reactive teleport system flings [owner] clear of [attack_text]!</span>")
+	playsound(get_turf(owner),'sound/magic/blink.ogg', 100, TRUE)
+	do_teleport(owner, get_turf(owner), tele_range, no_effects = TRUE, channel = TELEPORT_CHANNEL_BLUESPACE)
+	owner.rad_act(rad_amount)
+	reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
+	return TRUE
+
+/obj/item/clothing/suit/armor/reactive/teleport/emp_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>The reactive teleport system ALMOST flings [owner] clear of [attack_text], leaving something behind in the process!</span>")
+	var/drop_organ = prob(50)
+	if(drop_organ)
+		owner.spew_organ(0)
```

Comment:
bruh instant death on emp seems bad, especially when the rest of the armors emp effects are mildy annoying at worst

Reply:
Yeahhhhhh I dunno you may be right on this one. Tempted to go for the other option which is teleporting only the reactive armor

Reply:
I should say 50% of the time it's a limb which is way safer, and a good amount of the time it's a tongue, eye, or ear. something useless or at least not immediately  life threatening

Reply:
just make it do stun vomit, compared to getting shockd from tesla, being hit by potted plants of repulsive etc this just makes me want to trash bin it roundstart as rd

Reply:
> Tempted to go for the other option which is teleporting only the reactive armor

this was my idea and its funny